src/core/Object3D.js

```diff
@@ -835,5 +849,79 @@ Object3D.prototype = Object.assign( Object.create( EventDispatcher.prototype ),
 
 } );
 
+function Vector3WithDirtyFlag( x, y, z ) {
+
+	this._x = 0;
+	this._y = 0;
+	this._z = 0;
+
+	Vector3.call( this, x, y, z );
+
+	this._dirty = false;
+
+}
+
+Vector3WithDirtyFlag.prototype = Object.assign( Object.create( Vector3.prototype ), {
+
+	constructor: Vector3WithDirtyFlag,
+	isVector3WithDirtyFlag: true
+
+} );
+
+Object.defineProperties( Vector3WithDirtyFlag.prototype, {
+
+	x: {
+
+		get: function () {
+
+			return this._x;
+
+		},
+
+		set: function ( value ) {
+
+			this._x = value;
+			this._dirty = true;
+
+		}
+
+	},
+
+	y: {
+
+		get: function () {
+
+			return this._y;
+
+		},
+
+		set: function ( value ) {
+
+			this._y = value;
+			this._dirty = true;
+
+		}
+
+	},
+
+	z: {
+
+		get: function () {
+
+			return this._z;
+
+		},
+
+		set: function ( value ) {
+
+			this._z = value;
+			this._dirty = true;
+
+		}
+
+	}
+
+} );
+
```

Comment:
I don't make `QuaternionWithDirtyFlag` yet in this PR because not sure why but somehow using the extension of `Quaternion` seems to lead the bad performance on my Windows Chrome...